scripts/eachdist.py

```diff
@@ -573,18 +577,64 @@ def update_version_files(targets, version, packages):
         f'__version__ = "{version}"',
     )
 
+def update_dependency_requirements(deps, pkg, version):
+    updated_deps = []
+    for dep in deps:
+        req = Requirement(dep)
+        if req.name == basename(pkg):
+            specifiers_set = []
+            for spec in req.specifier:
+                if ".dev" in spec.version:
```

Comment:
Alternatively, to avoid repetition:

```
operator_symbols = {
    "<": "<",
    "<=": "<=",
    "==": "==",
    ">=": ">=",
    ">": ">",
    "!=": "!=",
    "~=": "~=",
}

for op, new_op in operator_symbols.items():
    if spec.operator == op and ".dev" in spec.version:
        specifiers_set.append(Specifier(f"{new_op}{spec.version.replace('.dev', '')}"))
```